samples/templates/default-azuredeploy.json

```diff
@@ -297,30 +359,46 @@
             ]
         },
         {
+            "condition": "[equals(parameters('solutionType'), 'FhirServerCosmosDB')]",
             "type": "Microsoft.KeyVault/vaults/secrets",
             "name": "[concat(variables('serviceName'), '/CosmosDb--Host')]",
             "apiVersion": "2015-06-01",
             "properties": {
                 "contentType": "text/plain",
-                "value": "[reference(concat('Microsoft.DocumentDb/databaseAccounts/', variables('serviceName'))).documentEndpoint]"
+                "value": "[if(equals(parameters('solutionType'), 'FhirServerCosmosDB'), reference(concat('Microsoft.DocumentDb/databaseAccounts/', variables('serviceName')), '2015-04-08').documentEndpoint, '')]"
```

Comment:
Do we need this condition here since condition above already check to make sure the solution type is cosmos db?

Reply:
Yes. The way that arm appears to do lookups is that it evaluates these regardless of the condition on the resource -- so without these checks you end up with not found exceptions.